### PR TITLE
[VTQueryServer] MySQLProxy drops transaction on error

### PR DESCRIPTION
**Pull request by danieltahara**
It's totally valid to have an error mid-transaction and then continue to act on the transaction. Unfortunately, the vtqueryserver mysql plugin handler drops the transaction ID (vis a vis ProxySession) on error.

Signed-off-by: Daniel Tahara <tahara@dropbox.com>

### CODE REVIEW

**Comment by danieltahara** [comment]
N.B. This is NOT ready. It's mostly an RFC.

**Comment by danieltahara** [comment]
Alright. Kept parity with vtgate, which always returns a session. Should be good to go.

File: go/vt/mysqlproxy/mysqlproxy.go [hunk 1]
```diff
@@ -78,7 +78,10 @@ func (mp *Proxy) Execute(ctx context.Context, session *ProxySession, sql string,
 	}
 
 	if err != nil {
-		return nil, nil, err
+		// N.B. You must return session here. Modeled after vtgate mysql plugin, the vtqueryserver
+		// plugin expects you to return a new or updated session and not drop it on the floor during
+		// an error.
+		return session, nil, err
```
Comment (sougou) [inline go/vt/mysqlproxy/mysqlproxy.go]: May be simpler to always `return session, result, err`?

Comment (danieltahara) [reply go/vt/mysqlproxy/mysqlproxy.go]: sure

**Comment by sougou** [comment]
Looks like vtqueryserver unit test needs to be fixed.

**Pull request merged**